src/app/_shared-components/MarkdownEditor/InitMdxEditor.tsx

```diff
@@ -62,6 +63,40 @@ const MAX_MENTION_SUGGESTIONS = 5;
 export default function InitializedMDXEditor({ editorRef, ...props }: { editorRef: ForwardedRef<MDXEditorMethods> | null } & MDXEditorProps) {
 	const t = useTranslations('Create.UploadImage');
 	const [openImageUploadDialog, setOpenImageUploadDialog] = useState(false);
+	const { quoteCommentText, setQuoteCommentText } = useQuoteCommentText();
+	const { theme } = useTheme();
+
+	// Handle quoted text insertion
+	useEffect(() => {
+		if (!quoteCommentText || !editorRef || typeof editorRef === 'function') {
+			return;
+		}
+
+		const editor = editorRef.current;
+		if (!editor) {
+			return;
+		}
+
+		const currentContent = editor.getMarkdown();
+
+		// Format the quoted text
+		const quotedText = `\n\n> ${quoteCommentText}\n\n`;
+
+		// Append to existing content
+		const newContent = currentContent + quotedText;
+
+		// Update the editor
+		editor.setMarkdown(newContent);
+
+		// Clear the quote text after using it
+		setQuoteCommentText('');
+
+		// Trigger onChange if provided
+		if (props?.onChange) {
+			props.onChange(newContent, false);
+		}
+	}, [quoteCommentText, editorRef, setQuoteCommentText, props]);
```

Comment:
only add the dependencies which are used inside the useEffect. don't have to pass whole props